File: 2001-2500/2125.Number of Laser Beams in a Bank.ts
function numberOfBeams(bank: string[]): number {
    let res = 0, prev = 0;
    for (const row of bank) {
        const cnt = [].filter.call(row, c => c == '1') .length;
        if (cnt) {
            res += prev * cnt;
            prev = cnt;
        }
    }
    return res;
};

File: 2001-2500/2125.Number of Laser Beams in a Bank.py
from collections import Counter

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        res = prev = 0
        for row in bank:
            cnt = Counter(row)
            if cnt['1']:
                res += prev * cnt['1']
                prev = cnt['1']
        return res

